pr_agent/algo/pr_processing.py

```diff
@@ -208,16 +208,32 @@ def pr_generate_compressed_diff(top_langs: list, token_handler: TokenHandler, mo
 
 
 async def retry_with_fallback_models(f: Callable):
+    # getting all models
     model = get_settings().config.model
     fallback_models = get_settings().config.fallback_models
     if not isinstance(fallback_models, list):
-        fallback_models = [fallback_models]
+        fallback_models = [m.strip() for m in fallback_models.split(",")]
     all_models = [model] + fallback_models
-    for i, model in enumerate(all_models):
+    # getting all deployments
+    deployment_id = get_settings().get("openai.deployment_id", None)
+    fallback_deployments = get_settings().get("openai.fallback_deployments", [])
+    if not isinstance(fallback_deployments, list) and fallback_deployments:
+        fallback_deployments = [d.strip() for d in fallback_deployments.split(",")]
+    if fallback_deployments:
+        all_deployments = [deployment_id] + fallback_deployments
```

Comment:
maybe it's better to validate that number of fallback deployments is equal to number of fallback models, and throw an exception otherwise

Reply:
Done, please see latest changes (also extracted the fallback parts to helper functions)